### ISSUE TITLE
Add support for running in a subdirectory.

### ISSUE THREAD

**Issue by jonhoo** [dungeon-revealer/dungeon-revealer#290]
I'd like to serve up the dungeon revealer web interface under a sub-path of a domain I control (through an nginx reverse proxy). Basically, I'd love to be able to have an nginx rule like
```
location /dnd-map/ {
    proxy_pass http://127.0.0.1:3000/;
}
```
Unfortunately, that won't currently work, since absolute paths are used for all external resources. For example, the stylesheet is linked at `/static/css/main.df0a185a.chunk.css`, but that ends up requesting `mydomain.com/static/css/main.df0a185a.chunk.css` rather than `mydomain.com/dnd-map/static/css/main.df0a185a.chunk.css`.

Is there a particular reason why absolute paths are used, and any chance that they could be converted to be relative to this use-case would work?

**Comment by n1ru4l** [comment]
I will have to dig deeper into this.

Other tools like Grafana, Prometheus Alert Manager and Co allow configuring a baseUrl that will be used. E.g. in your use-case the base url would be `http(s)://my-domain.com/dnd-map/`.

The HTML is currently built with create-react-app and served trough express. But I think it would definitely be possible to replace it based on the base url environment variable.

In case you are interested in implementing such a feature feel free to discuss and create a pr ☺️

**Comment by jonhoo** [comment]
I've worked around it for my use-case using a subdomain for the time being, so no urgent need, and I also have limited experience with react/express, otherwise I would :sweat_smile: Biggest downside of a subdomain is that it makes HTTPS support trickier.

**Comment by n1ru4l** [comment]
@jonhoo You probably already know it, but you could use an wildcard certificate.

**Comment by jonhoo** [comment]
Yeah, although they're trickier to get through Let's Encrypt, especially since my root domain is being served through CloudFlare (whereas my subdomains go directly to my backend server).

**Comment by sdsalyer** [comment]
Hey, thanks for working on this! Looking forward to trying out the change when it's released. :-)

**Comment by n1ru4l** [comment]
You guys could really help my out by testing this: https://github.com/dungeon-revealer/dungeon-revealer/pull/311

I added instructions here: https://github.com/dungeon-revealer/dungeon-revealer/pull/311#issuecomment-605497295
